File: 03.CodeFirst/01.OOPIntro/Person.cs
namespace _01.OOPIntro
{
    class Person
    {
        private string name;
        private int age;

        public int Age
        {
            get
            {
                return this.age;
            }

            set
            {
                this.age = value;
            }
        }

        public string Name
        {
            get
            {
                return this.name;
            }

            set
            {
                this.name = value;
            }
        }
        public Person(string name, int age)
        {
            this.name = name;
            this.age= age;
        }

        public Person() : this("No name", 1) { }
        public Person(int age) : this("No name", age) { }
        public Person(string name) : this(name, 1) { }

    }
}

File: JavaOOPBasics/05.Inheritance-Exercises/src/P01_Person/Person.java
package P01_Person;

class Person {
    protected String name;
    protected Integer age;

    Person(String name, Integer age) {
        setName(name);
        setAge(age);
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        if (name.length() >= 3) {
            this.name = name;
        } else {
            throw new IllegalArgumentException("Name's length should not be less than 3 symbols!");
        }
    }

    public Integer getAge() {
        return age;
    }

    public void setAge(Integer age) {
        if (age < 1) {
            throw new IllegalArgumentException("Age must be positive!");
        } else {
            this.age = age;
        }
    }

    @Override
    public String toString() {
        return String.format("Name: %s, Age: %d", this.getName(), this.getAge());
    }
}